### ISSUE TITLE
SDL3 feature request: SDL_JoystickConnectedFromInstanceID(SDL_JoystickID)

### ISSUE THREAD

**Issue by AntTheAlchemist** [libsdl-org/SDL#7100]
And if needed: SDL_GamepadConnectedFromInstanceID(SDL_JoystickID)

I am finding I often need to do this to avoid recursively opening gamepads:

``` 
if (!SDL_JoystickConnected(SDL_GetJoystickFromInstanceID(id))) {
// open joystick / gamepad
}
```

**Comment by AntTheAlchemist** [comment]
I was thinking because most of the other functions take a JoystickID, why doesn't SDL_JoystickConnected take an ID instead of a pointer to a joystick? If I already have a pointer, it's because I opened it, so I already know it's opened. I have to obtain a pointer first with `SDL_GetJoystickFromInstanceID(id)`, then test to see if it's opened (connected). If you see what I mean?

**Comment by AntTheAlchemist** [comment]
Maybe Add `SDL_JoystickOpened(SDL_JoystickID)` to test if it's already been opened, so there's no need to reach for a pointer first with `SDL_GetJoystickFromInstanceID`

**Comment by slouken** [comment]
Opened and connected are different things. Opened means you have an `SDL_Joystick *` somewhere that references it. Connected means it's still plugged into the computer.

**Comment by AntTheAlchemist** [comment]
@slouken I understand now, thank you. I wasn't aware `SDL_GetJoystickFromInstanceID` would return null if it hasn't been opened - that's useful - I've been using it wrong all this time. Can we add clarification about that to the documentation?

**Comment by slouken** [comment]
Sure, feel free to update the header documentation so it makes sense to you, and I'll review it for correctness. (or paste updated documentation here and I'll move it over to the header after review)

**Comment by AntTheAlchemist** [comment]
For `SDL_GetJoystickFromInstanceID` (and `SDL_GetGamepadFromInstanceID` will be similar)

```
/**
 * Get the SDL_Joystick associated with an instance ID, if it has been opened.
 *
 * \param instance_id the instance ID to get the SDL_Joystick for
 * \returns an SDL_Joystick on success or NULL on failure, or has not been opened yet; call SDL_GetError()
 *          for more information.
 *
 * \since This function is available since SDL 3.0.0.
 */
```

This should avoid the assumption (which I fell for) that a pointer will be returned even if it hasn't been opened yet.

**Comment by slouken** [comment]
Clarification added, thanks!
